### ISSUE TITLE
data science - match web to pdf colors in code snippets

### ISSUE THREAD

**Issue by srm-openstax** [openstax/cnx-recipes#5555]
Regarding the color snippets of the Python text in the web view: as per the Previous version PDF view, we expect the same layout and color to be applied in the WEB VIEW.
 
Please refer to the Previous version PDF view screenshot. The layout and color snippets in the web view need to be updated to match the PDF view.

**Previous Version(Sample):**
![Previous_version_PDF_View](https://github.com/openstax/cnx-recipes/assets/86622862/00e7067d-b033-4862-bc96-63ccb2a89fb5)

**PDF View:**
![PDF_View](https://github.com/openstax/cnx-recipes/assets/86622862/3e0350c6-d32f-49d1-aec9-682c382b4396)

**WEB View:**
![WEB_View](https://github.com/openstax/cnx-recipes/assets/86622862/a0100b03-9843-4881-a38e-23ef8ceae4a7)

# Issue Occurrence(s)

 **Location**
- [ ] In PDF
- [x] In Web

**Frequency**
- [ ] At every instance of the content
- [ ] Irregularly
- [ ] Other (Please describe below)

# Content Location(s)
- **osbooks repo:** https://github.com/openstax/osbooks-principles-data-science
- **collection slug:** Principles-data-science
- **ref/branch:** https://openstax-osbooksprincip-cvay7xh0p0y.ws-us110.gitpod.io/
- **chapter/section:** Chapter 03/3.5
- **surrounding text:** The round() function is then used to round the probability result to 3 decimal places.

# Acceptance Criteria

## Web
- Can we double check the CSS for web and see that the highlight.js classes get the same styling as the PDF (colorwise)? The hex values for code were developed with PDF and web in mind

**Comment by srm-openstax** [comment]
Larissa, I think we have two separate issues in this ticket. The web view color snippets do not match the PDF color. The

**Comment by kswanson33** [comment]
From investigation -- the web code color schema currently aligns with the python pdf code color schema.

Options:
1. (easiest) I change all web code to align with data science instead of python
2. Web code color schemas vary based on book (would require increased tagging)

I think I tag @openstaxalina about this -- what do you think?

**Comment by openstaxalina** [comment]
@kswanson33 unfortunately I think we will have to go with option 2 due to the colors being set from specific requests from the author team 😬 I will, however, verify this 100% with Larissa when I meet with her later. I'll update here with the final decision by EOD.

**Comment by openstaxalina** [comment]
@kswanson33 disregard my above comment -- I was confusing this with in text code discussed in another issue 😅 
 
Per https://github.com/openstax/cnx-recipes/issues/5348 and Larissa, the autocolored (by highlight.js) Python and Data Science code snippets should use all the same colors, in both web and PDF. Can you double-check where the discrepancy is happening?

**Comment by kswanson33** [comment]
Yep! So in the below screenshot, you can see that there are two flavors of "code keyword" colors, one generic and one for python.

![Image](https://github.com/openstax/cnx-recipes/assets/26280712/6cd895ef-4183-4987-a7e2-ab63aca56c0d)

Then, in data science, here are the color imports:

![Image](https://github.com/openstax/cnx-recipes/assets/26280712/30076614-fd74-4b05-9eef-b182195dff2e)

As you can see, generic colors are invoked (except the first one).

Whereas in webview, these are the color imports:

![Image](https://github.com/openstax/cnx-recipes/assets/26280712/5db4f4bd-4681-4b25-9c17-2c359ad3d3fe)

Webview uses the python colors.

**Comment by kswanson33** [comment]
@openstaxalina Does that help? Should I fix the data science PDF to match web instead of the other way around?

**Comment by openstaxalina** [comment]
@kswanson33 okay, I see - thanks so much for the explanation. Are you able to see when the `Python` versions of the keywords were set? I'm wondering why we didn't use the generic colors. I'm guessing/half-remembering y'all chose new ones for accessibility reasons, but I can't find evidence of that.

Also happy to huddle on this if easier 😅

**Comment by kswanson33** [comment]
[This](https://github.com/openstax/cnx-recipes/issues/4877) looks like the specification for the python (later) colors. Meanwhile, [this](https://github.com/openstax/cnx-recipes/issues/3750) is the card that set the original (earlier) colors. Interestingly enough, both of them seem to be intended to style code snippets in the python language.

Books that use the earlier colors: `computer-science`, `data-science`, `finance`

Books that use the later colors: `webview`, `python`

All books use the `cardboard` theme, except webview, which is its own theme.

SO yeah, I'm not sure what happened here either 😅 Iffff we want different color sets in different pdfs, and webview should always match pdf, that would be possible to achieve by tagging code snippets with `code-theme-tropical` or `code-theme-ocean` etc (or whatever you want). We could also pick one color set and stick with it.

**Comment by openstaxalina** [comment]
Thank you so much Kendra - this is very helpful. I am 99% sure we should make all books using highlight.js -- PDF and webview -- use the colors as outlined [here](https://github.com/openstax/cnx-recipes/issues/4877) for _Introduction to Python Programming_. 

I will confirm this 100% with Larissa at my meeting with her tomorrow. If that means this card slides to next sprint, that is fine since the book doesn't release for several months.

**Comment by openstaxalina** [comment]
@kswanson33 I did confirm 100% with Larissa that we should make all books using highlight.js -- PDF and webview -- use the colors as outlined https://github.com/openstax/cnx-recipes/issues/4877 for Introduction to Python Programming. Please let me know if you need anything further. I'm going to add the computer science label to this card so I remember to alert the team working on that title that these colors changed.

**Comment by kswanson33** [comment]
Hi @openstaxalina! I'm overwriting all the `hljs` colors from #3750 with the colors from #4877. As I look into this further, I'm seeing that `hljs` is not the only way code colors are tagged. As outlined in #3750, there are also the `.green-code`, `.red-code`, etc tags. 

I have only found one book, `finance`, on production with these tags in the content. The style for `computer-science` also supports these tags, but I don't see them in the content on any branch of the repo.

[Here](https://rex-web.herokuapp.com/books/052b8372-0c5e-4ff6-8fd3-326377e9e91f@f62a6fe/pages/1129068b-6268-42e8-a9ad-b181c1c220d5?archive=https://openstax.org/apps/archive/20240506.185246#sect-00002) is a page in `finance` with some code color tagged `span`s. 

![Image](https://github.com/openstax/cnx-recipes/assets/26280712/5f289eba-d6eb-4e8b-95e8-95f7b568e280)

As you can see, the colors have not been implemented on webview, but they are in the PDF, and the tags are in the content.

Additionally, some other books (`data-science`, `webview`) have just the `.green-code` tag without other colors. For this, a different green is used, `#006900` or G4 (as per #5500), rather than `#18775D` or G3 as per #3750.

My questions are:
1. Is it okay for `green-code` to refer to different greens depending on the book?
2. Are the other colors still supported?

Once I know, I'll update #3750 to note that either the entire card is archived or that the `hljs` colors are, & link to #4877.

**Comment by openstaxalina** [comment]
@kswanson33 thanks so much for the thorough update. I think we should standardize the greens -- and if I'm understanding correctly, this means _no change_ for the Data Science book because it is already using `006900` for `green-color` -- is this correct?

If so, let's keep this card specific to changes to Data Science, and I will make other cards for updating the "manual" text colors in the other books at a future time.

(The overall goal would be to have all colors unified -- if there's a green that `hljs` uses, that the manual color attribute would use the same HEX value. Let me know if this makes sense.)
